### PR TITLE
add initial migration

### PR DESCRIPTION
**Pull request by Genresal**

### CODE REVIEW

File: IVCRM.API/appsettings.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
         }
     },
     "ConnectionStrings": {
-        "Database": "Server=(local);Database=IvcrmDb;Trusted_Connection=True;Encrypt=False;"
+        "ConnectionString": "Server=(local);Database=IvcrmDb;Trusted_Connection=True;Encrypt=False;"
```
Comment (nantonov) [inline IVCRM.API/appsettings.json:9]: Not critical but start using MSSQLSERVER

File: IVCRM.BLL/IVCRM.BLL.csproj [hunk 2]
```diff
@@ -6,6 +6,11 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.10" />
```
Comment (nantonov) [inline IVCRM.BLL/IVCRM.BLL.csproj:10]: In BLL only Business => Business Logic Layer

File: IVCRM.BLL/ServiceCollectionRegistry.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+namespace IVCRM.BLL
+{
+    public static class ServiceCollectionRegistry
+    {
+        private const string ConnectionString = "ConnectionString";
+        public static void AddServices(this IServiceCollection services)
+        {
+        }
+
+        public static void AddEntityFrameworkSetup(this IServiceCollection services, IConfiguration configuration)
```
Comment (nantonov) [inline IVCRM.BLL/ServiceCollectionRegistry.cs:20]: Move to DAL

File: IVCRM.DAL/Entities/Customer.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace IVCRM.DAL.Entities
+{
+    public class Customer
```
Comment (nantonov) [inline IVCRM.DAL/Entities/Customer.cs:3]: Rename all entities on DAL like CustomerEntity (with postfix Entity)

File: IVCRM.DAL/Infrastructure/AppDbContext.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        public DbSet<Order> Orders { get; set; }
+        public DbSet<Product> Products { get; set; }
+        public DbSet<ProductCategory> ProductCategories { get; set; }
+        public DbSet<ProductOrder> ProductOrders { get; set; }
+        public DbSet<ProductStorage> ProductStorages { get; set; }
+        public DbSet<Storage> Storages { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Order>().HasOne(x => x.Customer).WithMany(x => x.Orders);
```
Comment (nantonov) [inline IVCRM.DAL/Infrastructure/AppDbContext.cs:23]: Fluent API is not necessary here, you can delete it to keep it clear

**Review by nantonov — CHANGES_REQUESTED** [review]

**Review by Blackolf931 — APPROVED** [review]

File: IVCRM.API/appsettings.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
         }
     },
     "ConnectionStrings": {
-        "Database": "Server=(local);Database=IvcrmDb;Trusted_Connection=True;Encrypt=False;"
+        "MSSQLSERVER": "Server=(local);Database=IvcrmDb;Trusted_Connection=True;Encrypt=False;"
```
Comment (nantonov) [inline IVCRM.API/appsettings.json:9]: I mean mssql server as database :)

**Review by nantonov — APPROVED** [review]
GJ 👍🏻

**Pull request merged**